circle/src/util.rs

```diff
@@ -0,0 +1,177 @@
+use p3_field::extension::Complex;
+#[cfg(test)]
+use p3_field::extension::ComplexExtendable;
+use p3_field::{ExtensionField, Field};
+#[cfg(test)]
+use p3_matrix::dense::RowMajorMatrix;
+#[cfg(test)]
+use p3_util::{log2_strict_usize, reverse_slice_index_bits};
+
+/// Get the cfft polynomial basis.
+/// The basis consists off all multi-linear products of: y, x, 2x^2 - 1, 2(2x^2 - 1)^2 - 1, ...
+/// The ordering of these basis elements is the bit reversal of the sequence: 1, y, x, xy, (2x^2 - 1), (2x^2 - 1)y, ...
+/// We also need to throw in a couple of negative signs for technical reasons.
+#[cfg(test)]
+pub(crate) fn circle_basis<F: ComplexExtendable>(point: Complex<F>, log_n: usize) -> Vec<F> {
+    if log_n == 0 {
+        return vec![F::one()];
+    }
+
+    let mut basis = vec![F::one()];
+    basis.reserve(1 << log_n);
+
+    // First compute the repeated applications of the squaring map π(x) = 2x^2 - 1
+    let mut cur = point.real();
+    for _ in 0..(log_n - 1) {
+        for i in 0..basis.len() {
+            basis.push(basis[i] * cur);
+        }
+        cur = F::two() * cur.square() - F::one();
```

Comment:
Is multiplication by two faster than addition with itself?

Reply:
Agreed addition may be preferable, there's also `.double()` to let the field impl pick what it thinks is best